node/api/api.go

```diff
@@ -162,3 +184,77 @@ func WriteJSON(w http.ResponseWriter, obj interface{}) {
 func WriteSuccess(w http.ResponseWriter) {
 	w.WriteHeader(http.StatusNoContent)
 }
+
+func GetPaginationDefaultRequest(req *http.Request) PaginationRequest {
+	return GetPaginationRequest(req, "start", "limit")
+}
+
+func GetPaginationRequest(req *http.Request, startQueryParam string, limitQueryParam string) PaginationRequest {
+	startString := req.FormValue(startQueryParam)
+	if startString == "" {
+		return PaginationRequest{
+			HasPaginationQuery:     false,
+			PaginationQueryIsValid: false,
+			Start: 0,
+			Limit: 0,
+		}
+	}
+	startIndex, err := strconv.Atoi(startString)
+	if err != nil || startIndex < 0 {
+		return PaginationRequest{
+			HasPaginationQuery:     true,
+			PaginationQueryIsValid: false,
+			Start: 0,
+			Limit: 0,
+		}
+	}
+	limit := DefaultPaginationSize
+	limitString := req.FormValue(limitQueryParam)
+	if limitString != "" {
+		limit, err = strconv.Atoi(limitString)
+		if err != nil || limit <= 0 {
+			return PaginationRequest{
+				HasPaginationQuery:     true,
+				PaginationQueryIsValid: false,
+				Start: 0,
+				Limit: 0,
+			}
+		}
+	}
+	return PaginationRequest{
+		HasPaginationQuery:     true,
+		PaginationQueryIsValid: true,
+		Start: startIndex,
+		Limit: limit,
+	}
+}
+
+// returns the starting index, end index, and total pages of a given slice for the page
+func GetPaginationIndicesAndResponse(sliceInterface interface{}, paginationRequest PaginationRequest) (PaginationWrapper, PaginationResponse) {
+	if paginationRequest.Limit <= 0 {
+		build.Critical("pagination request limit must be greater than 0")
+	}
+	slice := reflect.ValueOf(sliceInterface)
+	if slice.Kind() != reflect.Slice {
+		build.Critical("attempting to paginate on non-slice object type: ", slice.Kind())
+	}
+	startingPageIndex := int(math.Min(float64(paginationRequest.Start), float64(slice.Len())))
+	endingPageIndex := int(math.Min(float64(paginationRequest.Start+paginationRequest.Limit), float64(slice.Len())))
+	totalPages := slice.Len() / paginationRequest.Limit
+	if math.Mod(float64(slice.Len()), float64(paginationRequest.Limit)) != 0 {
```

Comment:
You probably don't need `math.Mod` with all the float casting. Instead you can do `if slice.Len() % paginationRequest.Limit != 0`